### PR TITLE
fix: including parentId when filter by eventTypeId happens

### LINKED ISSUE #10261: [CAL-2209] Insights, not counting managed event type bookings
Since managed event type bookings are special case when looking up specific eventTypeId, we should add to conditional to also look for parentId eventTypeId.

This means 2 things:

* No bookings will happen with managedEventTypeId
* Bookings will only count once when using parentId field.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2209](https://linear.app/calcom/issue/CAL-2209/insights-not-counting-managed-event-type-bookings)</sub>

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- When as admin/owner filters and select a eventTypeId that it's currently managed event type, count fails since using eventType filter will only look for specific id, now we add OR lookup for eventTypeId OR parentId.

Fixes #10261 

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Are there environment variables that should be set? No.
- What are the minimal test data to have? Having yarn seed-insights could help.
- What is expected (happy path) to have (input and output)? Managed event type bookings should be count when viewing team insights with a specific eventTypeId selected.
- Any other important info that could help to test that PR

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/insights/server/events.ts [hunk 1]
```diff
@@ -86,6 +86,17 @@ class EventsInsights {
...
+  static getTotalCompletedEvents = async (bookingIds: number[]) => {
+    return await prisma.bookingTimeStatus.count({
+      where: {
+        id: {
+          in: bookingIds,
+        },
+        timeStatus: "completed",
+      },
+    });
+  };
```
Comment (alannnc) [inline packages/features/insights/server/events.ts:98]: This functions it's not related to this problem but found out some irregularities when counting.

File: packages/features/insights/server/trpc-router.ts [hunk 2]
```diff
@@ -133,7 +133,14 @@ export const insightsRouter = router({
...
 
       if (eventTypeId) {
-        whereConditional["eventTypeId"] = eventTypeId;
+        whereConditional["OR"] = [
+          {
+            eventTypeId,
+          },
+          {
+            eventParentId: eventTypeId,
+          },
```
Comment (alannnc) [inline packages/features/insights/server/trpc-router.ts:142]: New conditional so we can get bookings that comes from a managed event Type.

File: packages/features/insights/server/trpc-router.ts [hunk 3]
```diff
@@ -257,7 +265,7 @@ export const insightsRouter = router({
           deltaPrevious: EventsInsights.getPercentage(baseBookings.length, lastPeriodBaseBookings.length),
         },
         completed: {
-          count: baseBookings.length - totalCancelled - totalRescheduled,
+          count: totalCompleted,
```
Comment (alannnc) [inline packages/features/insights/server/trpc-router.ts:268]: This is the fix for the irregularity found earlier.

File: packages/prisma/migrations/20230719214513_update_booking_time_status_fields_event_parent_id/migration.sql [hunk 4]
```diff
@@ -0,0 +1,35 @@
+-- View: public.BookingsTimeStatus
```
Comment (alannnc) [inline packages/prisma/migrations/20230719214513_update_booking_time_status_fields_event_parent_id/migration.sql:1]: New migration, views can only be created or replaced.

File: packages/prisma/schema.prisma [hunk 5]
```diff
@@ -877,20 +877,21 @@ model SelectedSlots {
...
+  createdAt     DateTime?
+  location      String?
+  paid          Boolean?
+  status        BookingStatus?
+  rescheduled   Boolean?
+  userId        Int?
+  teamId        Int?
+  eventLength   Int?
+  timeStatus    String?
+  eventParentId Int?
```
Comment (alannnc) [inline packages/prisma/schema.prisma:896]: Just added a field but changed everything.

**Review by zomars — APPROVED** [review]

**Review by zomars — CHANGES_REQUESTED** [review]
Migration failed to apply cleanly to the testing DB

image

**Comment by alannnc** [comment]
> Migration failed to apply cleanly to the testing DB
> 
> image

Just uploaded some changes, should be valid now.

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**